### ISSUE TITLE
[Bug]: Keyboard navigation doesn't work in Firefox

### ISSUE THREAD

**Issue by thecristen** [remix-run/react-router#10873]
### What version of React Router are you using?

6.16.0

### Steps to Reproduce

Go to the `<BrowserRouter />` example and try to click on any of the links using just the keyboard. 

### Expected Behavior

Pressing the <kbd>Tab</kbd> key will take me through each link.

### Actual Behavior

<kbd>Tab</kbd> skips all links in Firefox 117.0.1, rendering them completely inaccessible.

Seems to work as expected in Safari 16.6, haven't tried Chrome.

Not sure if it matters, but I see the same issue with `<NavLink />` too.

**Comment by thecristen** [comment]
Nevermind, forgot I was on a new computer and hadn't made the ritual of [clicking this checkbox](https://www.a11yproject.com/posts/macos-browser-keyboard-navigation/).
